midend/parserUnroll.cpp

```diff
@@ -565,7 +565,34 @@ class ParserSymbolicInterpreter {
             newStates.insert(newName);
         }
         for (auto s : state->state->components) {
-            auto* newComponent = executeStatement(state, s, valueMap);
+            const IR::StatOrDecl* newComponent = nullptr;
+            if (s->is<IR::IfStatement>()){
```

Comment:
This looks wrong to me. You should not try to hardwire some possible cases here, but rather generalize executeStatement to be recursive and handle blocks and ifs.

Reply:
Thanks for the advice, I'll do it.